src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -184,7 +184,8 @@ function validatePropTypes(element) {
 var ReactElementValidator = {
 
   createElement: function(type, props, children) {
-    var validType = typeof type === 'string' || typeof type === 'function';
+    var validType = typeof type === 'string' || typeof type === 'function' ||
+                    (type !== null && typeof type === 'object');
```

Comment:
This only changes the warning behavior, but it would be nice if we could be a little more strict here. Forgetting to export anything from a CommonJS module lands you an empty object here.